lib/src/Uncertainty/Algorithm/OrthogonalBasis/OrthogonalProductFunctionFactory.cxx

```diff
@@ -86,6 +87,13 @@ EnumerateFunction OrthogonalProductFunctionFactory::getEnumerateFunction() const
   return tensorizedFunctionFactory_.getEnumerateFunction();
 }
 
+/* Return the collection of univariate orthogonal polynomial families */
+OrthogonalProductFunctionFactory::FunctionFamilyCollection OrthogonalProductFunctionFactory::getFunctionFamilyCollection() const
+{
+  return coll_;
```

Comment:
Inside OrthogonalProductFunctionFactory, even if they are stored inside a collection of UniVariateFunctionFamily, you know that functions passed to tensorizedFunctionFactory_
are in fact OrthogonalUniVariateFunctionFamily, so IMO you can downcast them.